spring-cloud-stream-test-support/src/test/java/org/springframework/cloud/stream/test/aggregate/AggregateTestWithMain.java

```diff
@@ -0,0 +1,61 @@
+package org.springframework.cloud.stream.test.aggregate;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+import org.springframework.cloud.stream.aggregate.AggregateApplication;
+import org.springframework.cloud.stream.aggregate.AggregateApplicationBuilder;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.annotation.Transformer;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Marius Bogoevici
+ */
+public class AggregateTestWithMain {
+
+	@Test
+	public void testAggregateApplication() throws InterruptedException {
+		// emulate a main method
+		ConfigurableApplicationContext context = new AggregateApplicationBuilder().from(UppercaseProcessor.class)
+				.namespace("upper").to(SuffixProcessor.class).namespace("suffix").run();
+
+		AggregateApplication aggregateAccessor = context.getBean(AggregateApplication.class);
+		MessageCollector messageCollector = context.getBean(MessageCollector.class);
+		Processor uppercaseProcessor = aggregateAccessor.getBinding(Processor.class, "upper");
+		Processor suffixProcessor = aggregateAccessor.getBinding(Processor.class, "suffix");
+		uppercaseProcessor.input().send(MessageBuilder.withPayload("Hello").build());
+		Message<?> receivedMessage = messageCollector.forChannel(suffixProcessor.output()).poll(1, TimeUnit.SECONDS);
+		assertThat(receivedMessage).isNotNull();
+		assertThat(receivedMessage.getPayload()).isEqualTo("HELLO WORLD!");
+		context.close();
+	}
+
+	@Configuration
```

Comment:
They're in a different package and there's not much code to be shared, so I don't think that it would help a lot, but it instead it would make the test harder to read.